server/aesdsocket.c

```diff
@@ -186,197 +411,211 @@ int main(int argc, char *argv[])
         close(STDOUT_FILENO);
         close(STDERR_FILENO);  
 
-    }
+    }      
+
+ 
+    // Configure and start 10s timer
+    signal(SIGALRM, timer_handler);
```

Comment:
@SundarKrishnakumar consider using the threaded version of the timer so you can call non signal safe functions from it.